src/boot.lisp

```diff
@@ -281,6 +281,25 @@
                                   (list (first v) (third v))))
                            varlist)))))))
 
+(defmacro declare (&rest declarations)
```

Comment:
Declare isn't a macro actually.

We process declare in many forms (like lambdas).  If it is missing somewhere, you can use `parse-body` (defined in compiler.lisp) to extract declarations and optional docstring from a body.